### DISCUSSION TITLE
Appium for accessibility testing with Axe

### DISCUSSION THREAD

**Discussion by itkhanz** [appium/appium#20560] [Q&A]
Hi,
With increasing need for accessibility testing, many are required now to incorporate accessibility testing for mobile applications. We are using Appium primarily for our mobile automation needs together with Java Client and UiAutomator2, and XCUITest driver for native android and iOS apps.

For web applications, as per my knowledge there exist open source tools such as [Axe](https://github.com/dequelabs/axe-core) which many use to perform accessibility testing for Web. I checked it out and saw that Axe also has integrations for automated accessibility testing for Selenium Java which works with WebDriver to analyze web pages for accessibility:-

https://github.com/dequelabs/axe-core-maven-html
https://github.com/dequelabs/axe-core-maven-html/blob/develop/selenium/README.md

I found below solution which is not open source and is paid from [deque Axe](https://www.deque.com/axe/devtools/mobile-accessibility/):
https://docs.deque.com/devtools-mobile/2024.2.14/en/appium-example-java

I would like to know whether it is possible to use/adapt this open-source tool to work with Appium also using AppiumDriver to analyse native mobile screens? or perhaps there is some other way to use Appium for accessibility testing?

Any thoughts on this? I am also interested to learn from your experiences how other people are dealing with topics for mobile apps accessibility testing.

**Comment by mykola-mokhnach** [comment]
I'd say accessibility testing in general is not a simple topic, especially from the perspective when it's done for state organisations and related regulatory requirements. These requirements could be really complicated and differ from one country to another. Adjusting above requirements with actual validation rules is a tedious work, not even saying failing accessibility checks may lead to serious legal issues. 

That is why, as a company, I would rather prefer to rely on paid products there since they provide particular level of confidence, while open source ones are provided without any guarantees or support.
